src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -103,11 +103,14 @@ private[actions] abstract class RefreshActionBase(
   protected lazy val deletedFiles: Seq[String] = {
     val relation = previousIndexLogEntry.relations.head
     val originalFiles = relation.data.properties.content.fileInfos
-    val delFiles = (originalFiles -- currentFiles).map(_.name).toSeq
+    val delFiles = (originalFiles -- currentFiles).map(_.name)
+
+    // Remove duplicate deleted file names in the previous log entry.
+    val prevDeletedFiles = previousIndexLogEntry.deletedFiles.filterNot(delFiles.contains)
 
     // TODO: Add test for the scenario where existing deletedFiles and newly deleted
     //  files are updated. https://github.com/microsoft/hyperspace/issues/195.
-    delFiles ++ previousIndexLogEntry.deletedFiles
```

Comment:
Any reason not using `distinct` with the previous implementation? for example, `(delFiles ++ previousIndexLogEntry.deletedFiles).distinct`?

Reply:
Because delFiles is originally Set, so I wanted to utilize it.
It's okay to use `distinct` if you prefer it.